prefect_aws/ecs.py

```diff
@@ -701,9 +701,17 @@ def _watch_task_run(
         last_status = status = current_status
         t0 = time.time()
         while status != until_status:
-            task = ecs_client.describe_tasks(tasks=[task_arn], cluster=cluster_arn)[
+            tasks = ecs_client.describe_tasks(tasks=[task_arn], cluster=cluster_arn)[
                 "tasks"
-            ][0]
+            ]
+
+            if not tasks:
+                # Intermittently, the task will not be described
+                self.logger.debug(f"{self._log_prefix}: Task not found.")
+                time.sleep(self.task_watch_poll_interval)
+                continue
```

Comment:
Could we include a timeout check here to avoid running too long if the task is never found for some reason?

Reply:
Hm true we do bypass the timeout check, I'll move this around.